### ISSUE TITLE
Ensure kubectl explain works on service catalog resources

### ISSUE THREAD

**Issue by luksa** [kubernetes-incubator/service-catalog#1576]
Currently, all you get is this:
```
$ kubectl explain clusterservicebroker
error: group servicecatalog.k8s.io has not been registered

$ kubectl explain clusterservicebroker --api-version=servicecatalog.k8s.io/v1beta1
Error from server (NotFound): the server could not find the requested resource
```

**Comment by luksa** [comment]
#dibs

I've already analyzed this and have a few fixes ready. I'm filing this issue just so we can track everything (fix requires multiple changes in different projects)

**Comment by luksa** [comment]
While working on this, I've found the following problems:
- the Service Catalog API server does not expose the openapi schema yet (but this is solved by PR #1208)
- default RBAC rules don't allow the kube-aggregator to get the openapi schema from the Service Catalog API server (preparing PR now)
- kubectl unnecessarily complains about the servicecatalog.k8s.io group not being registered - you need to specify `--api-version`, but ideally that' shouldn't be necessary (preparing PR now)
- once the openapi schema is properly exposed to kubectl, it starts showing validation errors when creating Service Catalog resources (preparing PR now)

That's as much as I remember. I'll add additional things and link the PRs when ready.

**Comment by luksa** [comment]
I'm closing this as most of what's needed to get this working is now resolved. If you enable the serving of the OpenAPI schema in the Service Catalog API server, `kubectl explain` now works.
